--- a/include/clang/AST/AttrDump.inc	2018-10-01 07:27:53.570527761 -0700
+++ b/include/clang/AST/AttrDump.inc	2018-10-01 07:27:57.712527697 -0700
@@ -229,15 +229,9 @@
     break;
   }
   case attr::CPUDispatch: {
-    const auto *SA = cast<CPUDispatchAttr>(A);
-    for (const auto &Val : SA->cpus())
-      OS << " " << Val;
     break;
   }
   case attr::CPUSpecific: {
-    const auto *SA = cast<CPUSpecificAttr>(A);
-    for (const auto &Val : SA->cpus())
-      OS << " " << Val;
     break;
   }
   case attr::CUDAConstant: {
